hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/compact/AbstractGenericRecordBuilder.java

```diff
@@ -93,6 +93,10 @@ public GenericRecordBuilder setInt16(@Nonnull String fieldName, short value) {
     @Nonnull
     @Override
     public GenericRecordBuilder setGenericRecord(@Nonnull String fieldName, @Nullable GenericRecord value) {
+        if (value != null && !(value instanceof CompactGenericRecord)) {
+            throw new HazelcastSerializationException("You can only use compact generic records in a compact"
```

Comment:
I believe we need to capitalize some of these (compact, generic record etc.). Same for other places